lapce-data/src/data.rs

```diff
@@ -133,7 +136,13 @@ impl LapceData {
         let latest_release = Arc::new(None);
 
         let dirs: Vec<&PathBuf> = paths.iter().filter(|p| p.is_dir()).collect();
-        let files: Vec<&PathBuf> = paths.iter().filter(|p| p.is_file()).collect();
+        let mut files = paths.iter().filter(|p| p.is_file());
+
+        let parsed_paths: Vec<(PathBuf, LineCol)> = paths
+            .iter()
+            .filter_map(|p| Self::parse_path_line_column(p) )
+            .collect();
+
```

Comment:
1. `paths: Vec<PathBuf>` should be a pure `cli.paths` without any transformations or canonicalization, so go to `launch()` (https://github.com/lapce/lapce/blob/master/lapce-ui/src/app.rs#L139) and do necessary changes.
2. Make two vector variable:
```rust
let dirs = SmallVec::<[PathBuf; 3]>::new();
let files = = SmallVec::<[(PathBuf, Option<LineCol>); 3]>::new();
```
and collect (push to `dirs` and `files`) the the result of calling newly created `pub fn resolve_path<T: AsRef<Path>>(path: T) -> Option<PathType>` function, something like this:
```rust
paths.into_iter.for_each(|path| /* body of function */ );
```